### PR TITLE
CR-1120533 xgq: add application ID member to the XGQ cmd structure

### PR DESCRIPTION
**Pull request by rajkumar-xilinx**
XGQ new protocol introduced a new field called API_ID which
tells the VMR the actual request client is looking for.

Each API_ID value has fixed and dedicated API associated in VMR/VMC firmware.

By default, XGQ request for complete sensor data for any repo type request.

Signed-off-by: Rajkumar Rampelli <rajkumar@xilinx.com>

### CODE REVIEW

**Comment by rajkumar-xilinx** [comment]
I will add code owners once @xdavidz approves this PR.

**Review by maxzhen — APPROVED** [review]

File: src/runtime_src/core/include/xgq_cmd_vmr.h
Comment (xdavidz) [inline src/runtime_src/core/include/xgq_cmd_vmr.h:116]: We cannot directly change this without considering other commands. Currently the log_page and sensor_data are sharing exact same command payload.

Given the sensor_data has it's own spec, we will have to have a new payload for sensor_data itself.

So, again the change process would be:
1) design the spec first: https://confluence.xilinx.com/pages/viewpage.action?pageId=302345261#XGQ1.0CommandandQueueSpecification(WIP)
2) change the code on host
3) sync the code on VMR

Let's discuss this with a meeting.

8bits seems good range to me. but what the pid will be used for?

**Review by xdavidz — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/include/xgq_cmd_vmr.h
Comment (rajkumar-xilinx) [reply src/runtime_src/core/include/xgq_cmd_vmr.h:137]: pid value will be one of the values of "enum xgq_cmd_sensor_page_id" defined in vmr.h file.
It specifies sensor request type (*GET_SIZE, *BDINFO etc.)

**Comment by chvamshi-xilinx** [comment]
Hi @rajkumar-xilinx /@xdavidz . This PR has "do not merge" tag. Can we remove this tag and merge?

**Pull request merged**
